hack/lib/util.sh

```diff
@@ -139,6 +135,21 @@ kube::util::gen-doc() {
   ${cmd} "${tmpdir}"
   # create the list of generated files
   ls "${tmpdir}" | LC_ALL=C sort > "${tmpdir}/.files_generated"
+
+  # remove all old generated file from the destination
+  while read file; do
+    if [[ -e "${tmpdir}/${file}" && -n "${skipprefix}" ]]; then
+      local original=$(cat "${dest}/${file}" | grep -v "${skipprefix}")
+      local generated=$(cat "${dest}/${file}" | grep -v "${skipprefix}")
+      if [[ -z $(diff <(echo "${original}") <(echo "${generated}")) ]]; then
```

Comment:
You already have the two strings in memory and you don't really use the diff restuls... why not just the below?

```shell
  if [[ "${original}" == "${generated"} ]]l; then
    ...
```